### PR TITLE
feat(query): return error response and status code for remote query

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

Parse remote query error response and return RemoteQueryError object having status code, error type, and error message

### CODE REVIEW

File: query/src/main/scala/filodb/query/ResultTypes.scala [hunk 1]
```diff
@@ -19,6 +19,11 @@ final case class QueryError(id: String, t: Throwable) extends QueryResponse with
     t.getStackTrace.map(_.toString).mkString("\n")
 }
 
+final case class RemoteQueryError(id: String,
+                                  errorResponse: RemoteErrorResponse,
+                                  statusCode: Int) extends QueryResponse {
+  override def toString: String = s"RemoteQueryError id=$id RemoteErrorResponse=$errorResponse"
```
Comment (tjackpaul) [inline query/src/main/scala/filodb/query/ResultTypes.scala:25]: HTTP `StatusCode` is not needed in `toString` ?

File: query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala [hunk 2]
```diff
@@ -45,12 +45,24 @@ case class PromQlRemoteExec(queryEndpoint: String,
...
+
+    import PromCirceSupport._
+    import io.circe.parser
     remoteExecHttpClient.httpGet(queryContext.plannerParams.applicationId, queryEndpoint,
       requestTimeoutMs, queryContext.submitTime, getUrlParams())
       .map { response =>
-        response.unsafeBody match {
-          case Left(error) => QueryError(queryContext.queryId, error.error)
-          case Right(successResponse) => toQueryResponse(successResponse.data, queryContext.queryId, execPlan2Span)
+        if (response.body.isLeft)
```
Comment (tjackpaul) [inline query/src/main/scala/filodb/query/exec/PromQlRemoteExec.scala:54]: NiT: Open Parentheses in same line as `if` condition. Also please add a comment on why we check `.isLeft`

**Review by tjackpaul — APPROVED** [review]
lgtm.. thank you @TanviBhavsar

File: query/src/main/scala/filodb/query/ResultTypes.scala (hunk 1 above)
Comment (TanviBhavsar) [reply query/src/main/scala/filodb/query/ResultTypes.scala:25]: I forgot, will add

**Pull request merged**